src/Polly.Core/Strategy/OutcomeEvent.cs

```diff
@@ -78,16 +66,39 @@ public OutcomeEvent<TArgs> Register<TResult>(Func<Outcome<TResult>, TArgs, Value
     {
         Guard.NotNull(callback);
 
-        if (!_callbacks.TryGetValue(typeof(TResult), out var callbacks))
-        {
-            callbacks = new List<object> { callback };
-            _callbacks.Add(typeof(TResult), callbacks);
-        }
-        else
+        return ConfigureCallbacks<TResult>(c => c.Register(callback));
+    }
+
+    /// <summary>
+    /// Adds a callback for the specified result type.
+    /// </summary>
+    /// <typeparam name="TResult">The result type to add a callbacks for.</typeparam>
+    /// <param name="configure">Action that configures the callbacks.</param>
+    /// <returns>The current updated instance.</returns>
+    public OutcomeEvent<TArgs> ConfigureCallbacks<TResult>(Action<OutcomeEvent<TArgs, TResult>> configure)
+    {
+        Guard.NotNull(configure);
+
+        if (!_callbacks.ContainsKey(typeof(TResult)))
```

Comment:
Use `!TryGetValue()` here so we don't need to index on line 87?

Reply:
Done.